### ISSUE TITLE
Assocation .build clears amount if association called during setup

### ISSUE THREAD

**Issue by nitsujri** [RubyMoney/money-rails#648]
## TL;DR

If a model association is called during `.build` for setting `with_model_currency`, the `column_cents` gets reset to 0.

## Example

https://github.com/nitsujri/money-build-issue

To run it, load the `rails console` and run `User.run_example`. 

## Details 

It will run through showing `.build` works fine if we don't call the assocation, `user.currency`, during `with_model_currency`, but then if we do call an association, `amount_cents` becomes zero.

The problem [line is this](https://github.com/nitsujri/money-build-issue/blob/master/app/models/purchase.rb#L8). The only difference is a random `user.currency` that has no bearing on the final output.

After build, setting `amount` works just fine as we all would expect.

## Closing

This specifically caught us during a `accepts_nested_attributes_for` form. I left all the deprecations in, so the output is a bit messy.

I can also get a similar situation with `u=User.create; Purchase.new(user: u, amount: '222')` so it's build/new initializer? I'm just guessing at this point though.

## Thank you!

Thanks for this fantastic set of gems!

**Comment by tambakoo** [comment]
@semmons99 hi. This issue still persists for me. I'm on `money-rails (1.15.0)` . For me my `price_cents` gets reset to 0 in a nested attribute recored creation.

**Comment by websebdev** [comment]
I have the same issue, did anyone find a fix for that? Nested attributes ignore the `price` attribute, but it works with `price_cents`.
The `price` param will work if the order is being created or if the order item already exists. If the order exists and a new order item gets added, `price` will not work.

My workaround for now is to manually change the price to cents. For example:
```rb
order_params = params.require(:order).permit(
  :name,
  order_items_attributes: [ :id, :name, :quantity, :unit_price, :_destroy ]
)
if order_params[:order_items_attributes].present?
  order_params[:order_items_attributes].each do |key, order_item_params|
    order_item_params[:unit_price_cents] = Money.from_amount(order_item_params[:unit_price].to_f).cents
    order_item_params.delete(:unit_price)
    {
      key => order_item_params
    }
  end
end
order_params
```
